File: src/main/scala/uk/co/automatictester/wiremock/maven/plugin/util/ParameterUtil.scala
package uk.co.automatictester.wiremock.maven.plugin.util

import scala.collection.JavaConverters._
import scala.collection.mutable.MutableList

object ParameterUtil {

    private val RootDirParamPrefix: String = "--root-dir="

    def getAllParams(dir: String, nonDirParams: String): Array[String] = {
        val dirParams: List[String] = getDirParam(dir).toList
        val otherParams: List[String] = getNonDirParams(nonDirParams)
        val allParams: MutableList[String] = MutableList()
        allParams ++= dirParams
        allParams ++= otherParams
        allParams.asJava.toArray(Array.ofDim(1))
    }

    def getDirParam(dir: String): Array[String] = {
        val dirParam: List[String] = List(RootDirParamPrefix + dir)
        dirParam.asJava.toArray(Array.ofDim(1))
    }

    private def getNonDirParams(nonDirParams: String): List[String] = {
        nonDirParams.split(" ").toList
    }
}

File: src/main/java/uk/co/automatictester/wiremock/maven/plugin/util/ParameterUtil.java
package uk.co.automatictester.wiremock.maven.plugin.util;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class ParameterUtil {

    private ParameterUtil() {
    }

    private static final String ROOT_DIR_PARAM_PREFIX = "--root-dir=";

    public static String[] getAllParams(String dir, String nonDirParams) {
        List<String> dirParam = Arrays.asList(getDirParam(dir));
        List<String> otherParams = getNonDirParams(nonDirParams);
        List<String> allParams = new ArrayList<>();
        allParams.addAll(dirParam);
        allParams.addAll(otherParams);
        return allParams.toArray(new String[]{});
    }

    public static String[] getDirParam(String dir) {
        List<String> dirParam = new ArrayList<>();
        dirParam.add(ROOT_DIR_PARAM_PREFIX + dir);
        return dirParam.toArray(new String[]{});
    }

    private static List<String> getNonDirParams(String nonDirParams) {
        return getParamsFrom(nonDirParams);
    }

    private static List<String> getParamsFrom(String paramString) {
        String[] params = paramString.split(" ");
        return Arrays.asList(params);
    }
}
